gneiss/balances.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import division
+import numpy as np
+from skbio.stats.composition import clr_inv
+from collections import OrderedDict
+
+
+def _balance_basis(tree_node):
+    """ Helper method for calculating balance basis
+    """
+
+    n_tips = sum([n.is_tip() for n in tree_node.traverse()])
+    counts = _count_matrix(tree_node)
+    counts = OrderedDict([(x, counts[x])
+                          for x in counts.keys() if not x.is_tip()])
+    nds = counts.keys()
+    r = np.array([counts[n]['r'] for n in nds])
+    s = np.array([counts[n]['l'] for n in nds])
+    k = np.array([counts[n]['k'] for n in nds])
+    t = np.array([counts[n]['t'] for n in nds])
+
+    a = np.sqrt(s / (r*(r+s)))
+    b = -1*np.sqrt(r / (s*(r+s)))
+
+    basis = np.zeros((n_tips-1, n_tips))
+    for i in range(len(nds)):
+        basis[i, :] = np.array([0]*k[i] + [a[i]]*r[i] + [b[i]]*s[i] + [0]*t[i])
+    return basis, nds
+
+
+def balance_basis(tree_node):
+    """
+    Determines the basis based on tree
+
+    Parameters
+    ----------
+    treenode : skbio.TreeNode
+        Phylogenetic tree.  Must be a strictly bifurcating tree
```

Comment:
Does [this](https://github.com/biocore/gneiss/pull/3/files#diff-1079b10233352c49b5ec47ae9b0644e5R46) address your concerns?